tools/karl.cpp

```diff
@@ -1038,6 +1068,114 @@ void handle_arguments (int argc, char ** argv)
+      obPath.copy(argv_config_files[flag_cnt], obPath.size(), 0);
+      argv_config_files[flag_cnt][obPath.size()] = 0; // null string terminator 
+      flag_cnt++; 
+    } 
+
+    madara_logger_ptr_log(logger::global_logger.get (), logger::LOG_TRACE,
+                          "Found karl config file: %s\n", 
+                          full_path);                   
+       
+    // read each line of the text formatted file in, each line contains 
+    // one flag followed by a space then the param if there is a parameter
+    while (std::getline(file, flag_param))
+    {
+      flag = flag_param.substr(0, flag_param.find_first_of(" "));
+      if (flag_param.find_first_of(" ") == std::string::npos)
+      {
+        param = "";
+      }
+      else
+      {
+        param = flag_param.substr(flag_param.find_first_of(" ") + 1,
+        flag_param.length() - (flag_param.find_first_of(" ") + 1)); 
+      }      
+      madara_logger_ptr_log(logger::global_logger.get (), logger::LOG_TRACE,
+                            "Flag: %s, Param: %s\n", 
+                            flag, param);         
+      
+      if ((flag == "-cf" || flag == "--config-file") && file_cnt >= 0)
+      {
+        config_file = param;        
+        
+        madara_logger_ptr_log(logger::global_logger.get (), logger::LOG_TRACE,
+                              "Found user karl config file flag, param: %s, File Cnt: %s\n", 
+                              config_file, file_cnt);                  
+        file_cnt--;
+ 
+        return (load_config_file(argv_config_files, config_file, file_cnt, flag_cnt));        
+      }    
+      else 
+      {
+        if (file_cnt >= 0)
+        {
+          
+          madara_logger_ptr_log(logger::global_logger.get (), logger::LOG_TRACE,
+                                "Found a flag saving now...\n");                 
+          argv_config_files.push_back(nullptr);
+          argv_config_files[flag_cnt] = new char[flag.size() + 1];
+          flag.copy(argv_config_files[flag_cnt], flag.size(), 0);
+          argv_config_files[flag_cnt][flag.size()] = 0; // null string terminator
+          flag_cnt++;
+
+          if (param != "")
+          {
+            madara_logger_ptr_log(logger::global_logger.get (), logger::LOG_TRACE,
+                                "Found a param saving now...\n");             
+            argv_config_files.push_back(nullptr);
+            argv_config_files[flag_cnt] = new char[param.size() + 1];
+            param.copy(argv_config_files[flag_cnt], param.size(), 0); 
+            argv_config_files[flag_cnt][param.size()] = 0; // null string terminator         
+            flag_cnt++;
+          }
+
+        }      
+      }
+    }
+    file.close();
+    handle_arguments(argv_config_files.size(), &argv_config_files[0]);
+  
+    // garbage cleanup
+    for (unsigned int arg_index = 0; arg_index < argv_config_files.size(); arg_index++)
+    {
+      delete(argv_config_files[arg_index]); 
```

Comment:
I knew you would comment on this :) Yeah I wanted to use vector<string> object but noticed all conversion to char ** required a few lines or function call anyway then I went char* type to beget rid of the casts I was using. Thanks for the idea here I will see what i can do.

Reply:
On second thought, std::deque<std::string> would be best here, since that avoids copying when appended to. You'd then use a std::vector<char *> to make the char** to pass into `handle_arguments`

Reply:
Cool! Never saw that std type nice. Implementing...